thread_safe_multi.go

```diff
@@ -0,0 +1,31 @@
+package movavg
+
+import "sync"
+
+// ThreadSafe synchronizes access to Moving Average calculator.
+func ThreadSafeMulti(multiMA MultiMA) MultiMA {
+	return &ThreadSafeMultiMA{
+		multiMA: multiMA,
+	}
+}
+
+// ----------------------------------------------------------------------------
+
+type ThreadSafeMultiMA struct {
```

Comment:
Once you're returning `MultiMA` interface from `ThreadSafeMulti` constructor - there's no need to keep this struct public, it makes no sense, really:
- the struct is useless without constructor (fields are private)
- the struct doesn't have some additional methods to do anything, so it matches the `MultiMA` interface completely, without providing any extra methods